[SPARK-29476][WEBUI] add tooltip for Thread

PavithraRamachandran opened the pull request:
### What changes were proposed in this pull request?
Adding tooltip for Thread Dump - Thread Locks

Before:
![Screenshot from 2019-11-04 17-11-22](https://user-images.githubusercontent.com/51401130/68127349-b963f580-ff3b-11e9-8547-e01907382632.png)


After:
![Screenshot from 2019-11-13 18-12-54](https://user-images.githubusercontent.com/51401130/68768698-08e7a700-0649-11ea-804b-2eb4d5f162b4.png)


### Why are the changes needed?
Thread Dump tab do not have any tooltip for the columns, Some page provide tooltip , inorder to resolve the inconsistency and for better user experience.

### Does this PR introduce any user-facing change?
No

### How was this patch tested?
Manual

PavithraRamachandran committed:
tooltip for thread dump page

AmplabJenkins commented:
Can one of the admins verify this patch?

AmplabJenkins commented:
Can one of the admins verify this patch?

srowen commented:
These tooltips just don't add anything. "Thread ID" gets a tooltip like "ID of the thread". We just don't need tooltips here, and we don't need to add them where there is nothing else to say. I know there was an umbrella JIRA opened for all these pages but the right answer may be 'no action' in several cases.

PavithraRamachandran commented:
@srowen Tread Locks  could have tooltip i felt. The other columns are obvious. 
I couldnt phrase a proper tooltip for ThreadLocks. 
Do you suggest we have tooltip Thread Locks? If so could you help me phrasing the tooltip for the same?

srowen commented:
I could imagine just adding a tooltip for the 'locks' column like:

Objects whose lock the thread currently holds

PavithraRamachandran commented:
@srowen I have added tooltip only for locks .
I have added tooltip for search too.  What is your suggestion on the search tooltip? 
![Screenshot from 2019-11-12 18-01-21](https://user-images.githubusercontent.com/51401130/68671929-85f61c00-0576-11ea-8e5e-2d85beeeceff.png)

srowen commented on core/src/main/scala/org/apache/spark/ui/exec/ExecutorThreadDumpPage.scala:
```diff
@@ -76,7 +76,12 @@ private[ui] class ExecutorThreadDumpPage(
           <div class="form-group">
             <div class="input-group">
-              Search: <input type="text" class="form-control" id="search" oninput="onSearchStringChange()"></input>
+              <span data-toggle="tooltip" data-placement="top" title="Searches (case insensitive) for
```
Hm, I still think this might be noisy. Too many tooltips will start to interfere with usability of the screen. I think it's pretty obvious what search does, so I'm not sure this clarifies much.

PavithraRamachandran replied:
@srowen  i have placed tooltip only for locks. could you check

PavithraRamachandran commented:
@srowen  could you pls review this commit?

SparkQA commented:
**[Test build #4937 has started](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4937/testReport)** for PR 26386 at commit [`de26d88`](https://github.com/apache/spark/commit/de26d889e1bc218e478c32edf475f1096c078b50).

SparkQA commented:
**[Test build #4937 has finished](https://amplab.cs.berkeley.edu/jenkins/job/NewSparkPullRequestBuilder/4937/testReport)** for PR 26386 at commit [`de26d88`](https://github.com/apache/spark/commit/de26d889e1bc218e478c32edf475f1096c078b50).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

srowen commented:
Merged to master
